### PR TITLE
[CIVP-22369] - bug fix retry on iterate

### PR DESCRIPTION
**Pull request by byndcivilization**
fixed retry bug

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## Unreleased
+### Changed
+
+### Fixed
+- fixes bug whereby calls with iterate=True do not retry
```
Comment (mheilman) [inline CHANGELOG.md:9]: could you please put `(#413)` at the end of this?

**Review by mheilman — CHANGES_REQUESTED** [review]
Would you mind adding/revising a test so that the previous version would fail?

**Comment by byndcivilization** [comment]
@mheilman how does this look?

File: civis/tests/test_utils.py [hunk 2]
```diff
@@ -260,3 +288,75 @@ def test_no_retry_on_connection_error(mock_session):
...
+    api_response = {'key': 'value'}
+    session_context = mock_session.return_value.__enter__.return_value
+    session_context.send.return_value.json.return_value = api_response
+
+    session_context.send.return_value.status_code = 429
+    session_context.send.return_value.headers = {
+        'Retry-After': str(retry_after)
+    }
+
+    for verb in ['head', 'trace', 'get', 'put', 'options', 'delete', 'post']:
```
Comment (mheilman) [inline civis/tests/test_utils.py:343]: This function is the same as the one above except for the name and this line, right? Could you just put the lowercase verb strings in the list in the other function ([here](https://github.com/civisanalytics/civis-python/pull/413/files#diff-f2df2ec5c411a1d0c536fcfb23aa66d7196fa463abd9d22460617bb37d667db2R307))?  `pytest.mark.parametrize` would be another option, but that seems like overkill.

**Review by mheilman — APPROVED** [review]
LGTM

**Pull request merged**